lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -60,9 +60,10 @@ def supports?
       private_constant :FILTER
 
       def application_name(java_opts, credentials)
-        name = credentials['application-name'] || @configuration['default_application_name'] ||
-          @application.details['application_name']
+        name = ENV['APPDYNAMICS_APPLICATION_NAME'] || credentials['application-name']
```

Comment:
I agree. I think `cf set-env <app_name> JBP_CONFIG_APP_DYNAMICS_AGENT '[default_application_name: <custom-name>]'` should be the best way to handle this.